float32[4] Joints